plugins/dynamix.docker.manager/include/DockerClient.php

```diff
@@ -354,14 +355,47 @@ public function getIcon($Repository,$contName,$tmpIconUrl='') {
 			@copy($icon, $iconRAM);
 		}
 		if (!is_file($icon) && is_file($iconRAM)) {
-			@copy($iconRAM,$icon);
+			@copy($iconRAM, $icon);
 		}
 		if ( !is_file($iconRAM) ) {
-			exec("/usr/bin/logger ".escapeshellarg("$imageName: Could not download icon $imgUrl"));
+			exec("/usr/bin/logger ".escapeshellarg("$contName: Could not download icon $imgUrl"));
+		}
+		else {
+			$this->purgeUnusedIconFiles($contName, $iconFile);
 		}
 
 		return (is_file($iconRAM)) ? str_replace($docroot, '', $iconRAM) : '';
 	}
+
+	public function purgeUnusedIconFiles($contName, $keepIcon='') {
```

Comment:
This new function better handles the more complex removal process for this caching mechanism.